src/components/_app/Web3ConnectionManager/components/WalletSelectorModal/WalletSelectorModal.tsx

```diff
@@ -113,8 +107,16 @@ const WalletSelectorModal = ({ isOpen, onClose, onOpen }: Props): JSX.Element =>
   }, [keyPair, ready, router])
 
   const { data: shouldLinkToUser } = useSWR(
```

Comment:
sidequest: would be nice to extract it with `fetchShouldLinkToUser` into a `useShouldLinkToUser` hook so the logic is coupled there and not flooded here

Reply:
Sidequest completed :ayy:
https://github.com/agoraxyz/guild.xyz/pull/825/commits/22d533bef104baf0036f9ab7c018e15e12306689